testing/tests/DevExpress.core/utils.date.tests.js

```diff
@@ -521,6 +521,13 @@ QUnit.test("trimming time of date", function(assert) {
     assert.deepEqual(dateUtils.trimTime(date), new Date(2015, 7, 16), "Date is correct after time trimming");
 });
 
+QUnit.test("setting to the day end", function(assert) {
+    var date = new Date(2015, 7, 16, 2, 30);
+
+    assert.deepEqual(dateUtils.setToDayEnd(date).getHours(), 23, "Date is correct after time setting day end");
+    assert.deepEqual(dateUtils.setToDayEnd(date).getMinutes(), 59, "Date is correct after time setting day end");
```

Comment:
Whether setToDayEnd(new Date(2015, 7, 16, 2, 30)) should keep the passed [year/month/day] in its result value or can change these values in any way?

assert.deepEqual(dateUtils.setToDayEnd(new Date(2015, 7, 16, 2, 30)), new Date(2015, 7, 16, 23, 59), "setToDayEnd(new Date(2015, 7, 16, 2, 30))");

Reply:
And, whether or not it will manage these values correctly?

assert.deepEqual(dateUtils.setToDayEnd(new Date(2015, 7, 16, 0, 0, 0)), new Date(2015, 7, 16, 23, 59, 0, 0), "setToDayEnd(new Date(2015, 7, 16, 0, 0, 0))");

assert.deepEqual(dateUtils.setToDayEnd(new Date(2015, 7, 16, 0, 0, 1)), new Date(2015, 7, 16, 23, 59, 0, 0), "setToDayEnd(new Date(2015, 7, 16, 0, 0, 1))");

assert.deepEqual(dateUtils.setToDayEnd(new Date(2015, 7, 16, 23, 59, 1)), new Date(2015, 7, 16, 23, 59, 0, 0), "setToDayEnd(new Date(2015, 7, 16, 23, 59, 1))"); - here, the generated 'dayEnd' value will be less than the passed value ?